x-pack/plugins/graph/public/helpers/use_workspace_loader.ts

```diff
@@ -74,14 +74,43 @@ export const useWorkspaceLoader = ({
       store.dispatch({ type: 'x-pack/graph/RESET' });
     }
 
+    async function* pageThroughIndexPatterns() {
+      let page = 1;
+      let perPage = 1000;
+      let total = 0;
+      let savedObjects = [] as IndexPatternSavedObject[];
+
+      async function* makeRequest(): AsyncGenerator<IndexPatternSavedObject[]> {
+        await savedObjectsClient
+          .find<{ title: string }>({
+            type: 'index-pattern',
+            fields: ['title', 'type'],
+            perPage,
+            page,
+          })
+          .then((response) => {
+            perPage = response.perPage;
+            total = response.total;
+            page = response.page;
```

Comment:
so this function is starting `let with page = 1`, which is requesting the first page, then you assign the value `page = response.page;` which is I think also the first page, since it was requested. It seems to me that in the case when there is a bigger total number than 1000, it would trigger `makeRequest` forever, since page is not incremented?